Student/Student/Submissions/SubmissionButton/SubmissionButtonPresenter.swift

```diff
@@ -109,8 +110,16 @@ class SubmissionButtonPresenter: NSObject {
                 assignmentID: assignment.id
             ).getSessionlessLaunchURL { [weak self] url in
                 guard let url = url else { return }
-                let safari = SFSafariViewController(url: url)
-                self?.env.router.show(safari, from: view, options: .modal())
+                if assignment.isGoogleCloudAssignment {
+                    self?.env.router.show(
+                        GoogleCloudAssignmentViewController(url: url),
+                        from: view,
+                        options: .modal(embedInNav: true, addDoneButton: true)
+                    )
+                } else {
+                    let safari = SFSafariViewController(url: url)
+                    self?.env.router.show(safari, from: view, options: .modal())
+                }
```

Comment:
Have you considered moving this whole chunk of logic into `LTITools` ?  Thoughts?  Maybe some function that returns a ViewController that is the right kind.  You may have to pass in the assignment.??  Just a thought.

Reply:
That's a good idea. I'll give it a go.

Reply:
This lead me down a path of great refactoring. I moved all of "is this a google LTI" logic to LTITools since that seems like a better place. Thanks for the suggestion.